src/Components/Pokedex/Pokedex.js

```diff
@@ -1,47 +1,35 @@
-import React, { useState } from 'react'
-import PokeDetails from '../PokeDetails/PokeDetails'
-import './Pokedex.scss'
-import {getSinglePokemon} from '../../ApiCalls'
+import React, { useContext } from 'react';
+import PokeDetails from '../PokeDetails/PokeDetails';
+import './Pokedex.scss';
+import { AppContext } from '../../AppContext';
 
-const Pokedex = ({allPokemon}) => {
-  const [pokeDetails, setPokeDetails] = useState({})
+const Pokedex = ({ allPokemon, fetchSinglePokemon}) => {
+  const { singlePokemon, setSinglePokemon } = useContext(AppContext);
 
   const createSelectMenu = () => {
     const pokeNames = allPokemon.map((poke, index) => {
-     return  <option value={poke.url} key={index}>{poke.name}</option>
-    })
+      return  <option value={poke.url} key={index}>{poke.name}</option>;
+    });
     return (
       <select
         title="Pokemon List"
         className='pokemon-list'
-        onChange={(event) => {displaySinglePokemon(event)}}
+        onChange={(event) => { displaySinglePokemon(event); }}
       >
         {pokeNames}
       </select>
-    )
-  }
+    );
+  };
 
   const displaySinglePokemon = (event) => {
-    fetchSinglePokemon(event.target.value)
-  }
-
-
-  const fetchSinglePokemon = async (url) => {
-    try {
-      const singlePokemon = await getSinglePokemon(url)
-      setPokeDetails(singlePokemon)
-    } catch (error) {
-      console.log(error);
-    }
-
-  }
-
-
+    fetchSinglePokemon(event.target.value);
+  };
+  
   return (
     <section className="pokedex-outline">
       <section className="pokedex-screen">
         {createSelectMenu()}
-        {pokeDetails.name && <PokeDetails details={pokeDetails}/> }
+        {singlePokemon.name && <PokeDetails details={singlePokemon}/> }
```

Comment:
Are you intending for the Pokedex to reflect the current pokemon from the game? I think it's cool, reminds of scanning a pokemon in the show, but I want to make sure it's intended.

Reply:
And do you need Context for this? See my questions in `Game.js`